windows-terminal-quake/Program.cs

```diff
@@ -61,6 +46,47 @@ public static void Main(string[] args)
 			}
 		}
 
+		private static Process GetOrCreateWindowsTerminalProcess()
+		{
+			var process = Process.GetProcessesByName("WindowsTerminal").FirstOrDefault();
+
+			if (process == null)
+			{
+				process = new Process
+				{
+					StartInfo = new ProcessStartInfo
+					{
+						FileName = "wt",
+						UseShellExecute = false,
+						RedirectStandardOutput = true,
+						WindowStyle = ProcessWindowStyle.Maximized
+					}
+				};
+				process.Start();
+			}
+
+			if (process == null || process.HasExited)
+			{
+				throw new Exception("Can not ensure exited process is accessible");
+			}
+
+			try
+			{
+				// Note: Accessing mainWindowHandle already throws "Process has exited, so the requested information is not available."
+				if (process.MainWindowHandle == IntPtr.Zero)
+				{
+					throw new Exception("Can not access newly started process.");
+				}
+			}
+			catch (Exception)
+			{
+				process = Process.GetProcessesByName("WindowsTerminal").First();
+				process.Refresh();
+			}

```

Comment:
These lines used to be inside the if, 2 lines above, because it was only needed when the process was not already running before.

Reply:
FYI: fixed, with preservation of history in https://github.com/flyingpie/windows-terminal-quake/pull/22